### PR TITLE
[Storage] `az storage blob copy start`: migrate to track 2

### PR DESCRIPTION
**Pull request by zhoxing-ms**
**Description**  

`az storage blob copy start` migrate to track 2

**Testing Guide**  

Run `az storage blob copy start`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/__init__.py [hunk 1]
```diff
@@ -111,7 +111,15 @@ def register_path_argument(self, default_file_param=None, options_list=None):
 
     def register_source_uri_arguments(self, validator, blob_only=False):
         self.argument('copy_source', options_list=('--source-uri', '-u'), validator=validator, required=False,
-                      arg_group='Copy Source')
+                      arg_group='Copy Source',
+                      help='A URL of up to 2 KB in length that specifies an Azure file or blob. '
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/__init__.py:115]: please adjust the help message. you need to add different help message for different command to be consistent with track1 interface. For example, `az storage blob copy start` vs `az storage blob incremental-copy start`

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/__init__.py:115]: also need to add backtick for https url for document issue.

File: src/azure-cli/azure/cli/command_modules/storage/operations/blob.py [hunk 2]
```diff
@@ -641,3 +641,43 @@ def set_blob_tier_v2(client, container_name, blob_name, tier, blob_type='block',
...
+
+    source_modified_access_conditions = cmd.get_models('_generated.models._models#SourceModifiedAccessConditions',
+                                                       resource_type=ResourceType.DATA_STORAGE_BLOB)
+    source_modified_access_conditions.source_if_match = source_if_match
+    source_modified_access_conditions.source_if_modified_since = source_if_modified_since
+    source_modified_access_conditions.source_if_none_match = source_if_none_match
+    source_modified_access_conditions.source_if_unmodified_since = source_if_unmodified_since
+
+    kwargs = {
+        'modified_access_conditions': modified_access_conditions,
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/operations/blob.py:671]: I am not sure if there is misunderstanding, but from SDK side I haven't seen the `modified_access_conditions` in keyword. 
But it works as expected, that's should be fine.

File: src/azure-cli/azure/cli/command_modules/storage/__init__.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/storage/__init__.py]: Good catch!  Actually, this example doesn't need to be added here. It already exists in the `_help` file, so I deleted it~

**Pull request closed without merging**